### PR TITLE
custom data management (for multiple tenant Kafka)

### PR DESCRIPTION
**Pull request by localvar**

### CODE REVIEW

File: pkg/api/customdata.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"go.etcd.io/etcd/client/v3/concurrency"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	// CustomDataPrefix is the object prefix.
+	CustomDataPrefix = "/customdata/{kind}"
```
Comment (samutamm) [inline pkg/api/customdata.go:30]: Here `CustomDataPrefix = /customdata/{kind}` but in pkg/cluster/layout.go there is `customDataPrefixFormat = /custom-data/%`. Is it possible to have only one variable for this?

Comment (localvar) [reply pkg/api/customdata.go:30]: unfortunately, we are not able to do this, the `{kind}` in `CustomDataPrefix` is for the `chi` framework to extract parameters in the request URL, '%s` in `customDataPrefixFormat` is for `fmt.Sprintf`.

**Review by samutamm — APPROVED** [review]

**Comment by haoel** [comment]
@localvar please add some documents for this feature.

**Comment by localvar** [comment]
This feature is ready for the requirement of multiple tenant Kafka now, but the idea of this feature is the same as the CustomResource feature in EaseMesh. So I will do a refactor to avoid code duplication, but this may introduce some changes, so I plan to write the document after the refactor, is this fine?

**Comment by haoel** [comment]
fine. you can include the documents in the refactor.

**Review by suchen-sci — APPROVED** [review]

**Pull request merged**